string-manipulation-3/src/title-case.js

```diff
@@ -0,0 +1,239 @@
+ * 95. then use an if statement to check if the currentWord string ends with a colon;
+ * 96. if it does, assign the boolean value of true to the colon variable;
+ * 97. then after the end of that if statement, assign the boolean value of true to the dealtWith variable;
+ * 98. then after the end of that if statement, concatenate the titleCased string with the currentWord string, and then:
+ * 99. concatonate the result of that expression with an empty space, and then assign the result of that expression back to the titleCased variable;
+ * 100. then after the for loop, trim the end of the titleCased string and assign it back to the titleCased variable;
+ * 101. return from the function the value of the titleCased string;
+ */
+
+function titleCase(title) {
+  var lowerCased = title.toLowerCase();
+  var titleWords = lowerCased.split(' ');
+  var colon = false;
+  var dealtWith = false;
+  var capitalLetter = '';
+  var titleCased = '';
+  var currentWord = '';
+
+  for (var i = 0; i < titleWords.length; i++) {
+    currentWord = titleWords[i];
+    dealtWith = false;
+    if (currentWord.includes('-')) {
+      var wordsArray = currentWord.split('-');
+      var localWord = '';
+      currentWord = '';
+      for (var j = 0; j < wordsArray.length; j++) {
+        dealtWith = false;
+        localWord = wordsArray[j];
+        switch (localWord) {
+          case 'javascript':
+            localWord = 'JavaScript';
+            colon = false;
+            dealtWith = true;
+            break;
+          case 'javascript:':
+            localWord = 'JavaScript:';
+            colon = true;
+            dealtWith = true;
+            break;
+          case 'api':
+            localWord = 'API';
+            colon = false;
+            dealtWith = true;
+            break;
+          case 'api:':
+            localWord = 'API:';
+            colon = true;
+            dealtWith = true;
+        }
+        if (!dealtWith) {
+          colon = false;
+          capitalLetter = localWord.at(0);
+          lowerCased = localWord.slice(1);
+          localWord = capitalLetter.toUpperCase() + lowerCased;
+          if (localWord.endsWith(':')) {
+            colon = true;
+          }
+          dealtWith = true;
+        }
+        currentWord += localWord + '-';
+      }
+      currentWord = currentWord.slice(0, currentWord.length - 1);
+    }
+    if (!dealtWith) {
+      switch (currentWord) {
+        case 'javascript':
+          currentWord = 'JavaScript';
+          colon = false;
+          dealtWith = true;
+          break;
+        case 'javascript:':
+          currentWord = 'JavaScript:';
+          colon = true;
+          dealtWith = true;
+          break;
+        case 'api':
+          currentWord = 'API';
+          colon = false;
+          dealtWith = true;
+          break;
+        case 'api:':
+          currentWord = 'API:';
+          colon = true;
+          dealtWith = true;
+      }
```

Comment:
This behavior has been repeated more than once, I would make it a helper function you can call within this function.

Reply:
my function keep breaking whenever I try making this a function...